presto-spark-base/src/main/java/com/facebook/presto/spark/execution/property/WorkerProperty.java

```diff
@@ -22,17 +22,58 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
-/**
- * A utility class that helps with properties and its materialization.
- */
-public class WorkerProperty
+public abstract class WorkerProperty<T1 extends NativeExecutionConnectorConfig, T2 extends NativeExecutionNodeConfig, T3 extends NativeExecutionSystemConfig>
 {
-    private WorkerProperty()
+    private final T1 connectorConfig;
+    private final T2 nodeConfig;
+    private final T3 systemConfig;
+
+    public WorkerProperty(
+            T1 connectorConfig,
+            T2 nodeConfig,
+            T3 systemConfig)
+    {
+        this.systemConfig = requireNonNull(systemConfig, "systemConfig is null");
+        this.nodeConfig = requireNonNull(nodeConfig, "nodeConfig is null");
+        this.connectorConfig = requireNonNull(connectorConfig, "connectorConfig is null");
+    }
+
+    public T3 getSystemConfig()
+    {
+        return systemConfig;
+    }
+
+    public T2 getNodeConfig()
+    {
+        return nodeConfig;
+    }
+
+    public T1 getConnectorConfig()
+    {
+        return connectorConfig;
+    }
+
+    public void populateSystemProperties(Path path)
+            throws IOException
+    {
+        populateProperty(systemConfig.getAllProperties(), path);
+    }
+
+    public void populateNodeProperties(Path path)
+            throws IOException
+    {
+        populateProperty(nodeConfig.getAllProperties(), path);
+    }
+
+    public void populateConnectorProperties(Path path)
```

Comment:
Can we have a single (making up a name here) populatePropertyFiles() method that populates all properties? Subclass can override. It'll probably be cleaner that way.